shared/src/main/java/org/odk/collect/shared/locks/ChangeLock.kt

```diff
@@ -9,22 +9,23 @@ interface ChangeLock {
      * function return value will be returned from the call to this method.
      */
     fun <T> withLock(function: Function<Boolean, T>): T {
-        val acquired = tryLock(defaultToken)
+        val acquired = tryLock(Token.DEFAULT)
 
         return try {
             function.apply(acquired)
         } finally {
             if (acquired) {
-                unlock(defaultToken)
+                unlock(Token.DEFAULT)
             }
         }
     }
 
-    fun tryLock(token: Any): Boolean
+    fun tryLock(token: Token): Boolean
 
-    fun unlock(token: Any)
+    fun unlock(token: Token)
+}
 
-    companion object {
-        val defaultToken = Any()
-    }
+enum class Token {
```

Comment:
I get that there are some advantages to using an `enum` here, but I don't think `ChangeLock` should have to know anything about "form entry" (or any other lock token specifics). I like the previous `Any` based implementation you had - I think you could just have the `FORM_ENTRY` token be a static object within `org.odk.collect.android.formentry`.